src/services/reports.js

```diff
@@ -204,8 +218,12 @@ module.exports = class ReportsHelper {
 				result: reportDataResult,
 			})
 		} catch (error) {
-			console.error('Error in getReportData:', error)
-			throw error
+			return responses.failureResponse({
+				message: 'INTERNAL_SERVER_ERROR',
+				statusCode: httpStatusCode.internal_server_error,
+				responseCode: 'SERVER_ERROR',
+				error: error.message,
+			})
```

Comment:
In [generateAndUploadCSV](https://github.com/ELEVATE-Project/mentoring/blob/49135d03c78918057615556e4f9695502275ae78/src/services/reports.js#L233), the temporary file created during CSV generation is not being removed, which could lead to unnecessary file accumulation. Can we add logic to ensure the file is deleted after the upload is successful, and also handle clean-up in case the function encounters an error?